core/src/main/java/org/apache/calcite/sql/SqlCall.java

```diff
@@ -146,6 +146,9 @@ public int operandCount() {
       return litmus.fail("{} != {}", this, node);
     }
     SqlCall that = (SqlCall) node;
+    if (!equalDeep(this.getFunctionQuantifier(), that.getFunctionQuantifier(), litmus)) {
+      return litmus.fail("{} != {} (functionQualifier differs)", this, node);
```

Comment:
minor: message says "functionQualifier", and I guess it should be "function quantifier".
Also, personally I would prefer to add this new check after the operator's name check (i.e. reach the verification of function quantifier only in those cases where operator names' match).